core/src/jsMain/kotlin/Transaction.kt

```diff
@@ -158,37 +158,78 @@ public open class WriteTransaction internal constructor(
 ) : Transaction(transaction) {
 
     /**
-     * Adds a new item to the database. If an item with the same key already exists, this will fail.
+     * Adds a new item to the database using an in-line or auto-incrementing key. If an item with the same
+     * key already exists, this will fail.
      *
      * This API is delicate. If you're passing in Kotlin objects directly, you're probably doing it wrong.
      *
      * Generally, you'll want to create an explicit `external interface` and pass that in, to guarantee that Kotlin
      * doesn't mangle, prefix, or otherwise mess with your field names.
      */
-    public suspend fun ObjectStore.add(item: dynamic) {
+    public suspend fun ObjectStore.add(item: dynamic): dynamic {
         val request = objectStore.add(item)
-        request.onNextEvent("success", "error") { event ->
+        return request.onNextEvent("success", "error") { event ->
             when (event.type) {
                 "error" -> throw ErrorEventException(event)
-                else -> Unit
+                else -> request.result
+            }
+        }
+    }
+
+    /**
+     * Adds a new item to the database using an explicit out-of-line key. If an item with the same key already
+     * exists, this will fail.
+     *
+     * This API is delicate. If you're passing in Kotlin objects directly, you're probably doing it wrong.
+     *
+     * Generally, you'll want to create an explicit `external interface` and pass that in, to guarantee that Kotlin
+     * doesn't mangle, prefix, or otherwise mess with your field names.
+     */
+    public suspend fun ObjectStore.add(item: dynamic, key: Key): dynamic {
```

Comment:
Is there perhaps a benefit to arranging the parameters w/ key first? To be familiar (similar) to collection methods?

Reply:
> same order as the original JS function

Makes sense, although I imagine a majority of library consumers will be Kotlin first.
We're also pre-1.0.0 at this point, so now is the time to really break things.
Fine with leaving it as-is, just pointing out that it might be better to make it `key` parameter first now, rather than later?